File: Scripts/QueryScripts/CountryDistribution.R
source("/Users/ricardocunha/Documents/FEUP/5ano/MSc\ Thesis/Thesis/Code/Scripts/Utils/mainBody.R")

countries <- vector()
nQueries <- vector()
nTotalQueries <- 0


parseString <- function(str){
  #print(str)
  #test for key presence
  if(str != "EMPTY"){
    i <- 1
    while(i<=length(countries)){
      if(countries[i] == str){
        nQueries[i] <<- nQueries[i]+1
        nTotalQueries <<- nTotalQueries+1
        return()
      }
      i <- i+1
    }
    
    countries <<- append(countries, str)
    nQueries <<- append(nQueries, 1)
    nTotalQueries <<- nTotalQueries+1
  }
  
}

printResults <- function(filename){
  i<-1
  write("QUERIES PER COUNTRY", file=filename, append=FALSE)
  while(i<=length(countries)){
    write(paste(countries[i],nQueries[i],percent(nQueries[i]/nTotalQueries),sep=":"),
          file=filename,append=TRUE)
    i <- i+1
  }
  
}

main <- function(){
  readFromStdin()
  filename <- "/Users/ricardocunha/Documents/FEUP/5ano/MSc\ Thesis/Thesis/Code/OutputFiles/CountryDistribution.txt"
  printResults(filename)
}

main();

File: Scripts/QueryScripts/CountryDistribution.pl
#!/usr/bin/perl -w
BEGIN { push @INC, '/Users/ricardocunha/Documents/FEUP/5ano/MSc Thesis/Thesis/Code/Scripts/Utils' }
require "sessionFileIndexes.pl";
require "checkDeviceType.pl";

$totalQueries=0;
%hash = ();

sub isCountryOnHash{
    $country = shift;
    return exists($hash{$country});
}

sub addCountry{
    $country = shift;
    if(isCountryOnHash($country)){
        $hash{$country}++;
    } else {
        $hash{$country} = 1;
    }
    
}

sub printHash{
    $counter=0;
    print "$totalQueries\n";
    foreach $key (sort{$hash{$b} <=> $hash{$a}} keys %hash){
        if($counter<10){
            print "$key>>>$hash{$key}\n";
            $counter++;
        } else {
            return;
        }
    }
}

sub main{
    while($_ = <STDIN>){
        @query = split(/>>>/, $_);
        chop($query[$countryIndex]);
        if($query[$countryIndex] ne "EMPTY"){
            $totalQueries++;
            addCountry($query[$countryIndex]);
        }
    }
    
    printHash;
}

main;



